test/tests/treeview-1a.js

```diff
@@ -0,0 +1,675 @@
+'use strict';
+
+const { ariaTest } = require('..');
+const { By, Key } = require('selenium-webdriver');
+const assertAttributeValues = require('../util/assertAttributeValues');
+const assertAriaLabelledby = require('../util/assertAriaLabelledby');
+const assertRovingTabindex = require('../util/assertRovingTabindex');
+
+const exampleFile = 'treeview/treeview-1/treeview-1a.html';
+
+const ex = {
+  treeSelector: '#ex1 [role="tree"]',
+  treeitemSelector: '#ex1 [role="treeitem"]',
+  groupSelector: '#ex1 [role="group"]',
+  folderSelector: '#ex1 [role="treeitem"][aria-expanded]',
+  topLevelFolderSelector: '#ex1 [role="tree"] > [role="treeitem"]',
+  nextLevelFolderSelector: '[role="group"] > [role="treeitem"][aria-expanded]',
+  docSelector: '#ex1 .doc[role="treeitem"]',
+  textboxSelector: '#ex1 #last_action'
+};
+
+const openAllFolders = async function (t) {
+  const closedFoldersSelector = ex.treeitemSelector + '[aria-expanded="false"]';
+  let folders = await t.context.session.findElements(By.css(closedFoldersSelector));
+
+  // While their are closed folders
+  while (folders.length != 0) {
+
+    // Find the first visible closed folder and click it
+    for (let folder of folders) {
+      if (await folder.isDisplayed()) {
+        await folder.click();
+        break;
+      }
+    }
+
+    folders = await t.context.session.findElements(By.css(closedFoldersSelector));
+  }
+};
+
+const checkFocus = async function (t, selector, index) {
+  return await t.context.session.executeScript(function (/* selector, index*/) {
+    const [selector, index] = arguments;
+    const items = document.querySelectorAll(selector);
+    return items[index] === document.activeElement;
+  }, selector, index);
+};
+
+const checkFocusOnParentFolder = async function (t, el) {
+  return await t.context.session.executeScript(function () {
+    const el = arguments[0];
+
+    // the element is a folder
+    if (el.hasAttribute('aria-expanded')) {
+      return document.activeElement == el.parentElement.closest('[role="treeitem"][aria-expanded]');
+    }
+    // the element is a folder
+    else {
+      return document.activeElement == el.closest('[role="treeitem"][aria-expanded]');
+    }
+  }, el);
+};
+
+const isTopLevelFolder = async function (t, el) {
+  return await t.context.session.executeScript(function () {
+    const el = arguments[0];
+    return el.parentElement.getAttribute('role') == 'tree';
+  }, el);
+};
+
+const isFolderTreeitem = async function (el) {
+  return !(await el.getAttribute('class')).includes('doc');
+};
+
+const isOpenedFolderTreeitem =  async function (el) {
+  return await el.getAttribute('aria-expanded') === 'true';
```

Comment:
Async functions are already wrapped in a `Promise.resolve`, so there's no need to `return await` here (or similar). And since I've never actually tested this, it looks like `return await` actually adds some overhead:

```javascript
const first = async () => await 'first';
const second = async () => 'second';
first().then(console.log);
second().then(console.log);
// second
// first
```